cmd/kube-apiserver/app/options/options.go

```diff
@@ -49,7 +50,19 @@ func NewAPIServer() *APIServer {
 		ServerRunOptions: genericoptions.NewServerRunOptions().WithEtcdOptions(),
 		EventTTL:         1 * time.Hour,
 		KubeletConfig: kubeletclient.KubeletClientConfig{
-			Port:        ports.KubeletPort,
+			Port: ports.KubeletPort,
+			PreferredAddressType: []string{
```

Comment:
Why use `[]string` here, just leave them as the `api` type until you need to compare?